### PR TITLE
✨ Refactor : Refactor Table Entity Conversation and Task Flow Logic  #5734

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
### Describe your changes :

I worked on the issue #5734 for table entity

#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Frontend Preview (Screenshots) :

**With Permission**

https://user-images.githubusercontent.com/59080942/176403507-a2d04b5d-783e-423f-9a2c-5d42ebdac993.mov

**Without Permission**

https://user-images.githubusercontent.com/59080942/176403646-26c80aac-b310-4c9a-ae51-42042c039643.mov

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/ActivityFeed/ActivityThreadPanel/ActivityThreadPanel.tsx [hunk 1]
```diff
@@ -37,6 +39,12 @@ const ActivityThreadPanel: FC<ActivityThreadPanelProp> = ({
     setActiveTab(key);
   };
 
+  useEffect(() => {
+    if (isEqual(threadType, ThreadType.Conversation)) {
+      setActiveTab('2');
```
[the remark this answers is not in the document]
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/ActivityFeed/ActivityThreadPanel/ActivityThreadPanel.tsx:44]: @ShaileshParmar11, Yes, we can create an enum and use it here.

File: openmetadata-ui/src/main/resources/ui/src/components/DatasetDetails/DatasetDetails.component.tsx [hunk 2]
```diff
@@ -647,6 +659,10 @@ const DatasetDetails: React.FC<DatasetDetailsProps> = ({
                         FQN_SEPARATOR_CHAR
                       )}
                       columns={columns}
+                      entityFieldTasks={getEntityFieldThreadCounts(
+                        'columns',
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/DatasetDetails/DatasetDetails.component.tsx:663]: same here -> const/enum

File: openmetadata-ui/src/main/resources/ui/src/utils/CommonUtils.tsx [hunk 3]
```diff
@@ -631,3 +637,40 @@ export const getExploreLinkByFilter = (
...
+        err,
+        jsonData['api-error-messages']['fetch-entity-feed-count-error']
+      );
+    });
+
+  getFeedCount(getEntityFeedLink(entityType, entityFQN), ThreadType.Task)
+    .then((res: AxiosResponse) => {
+      entityCallback(res.data.totalCount);
+      taskCallback(res.data.counts);
+    })
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/utils/CommonUtils.tsx:669]: same here, for empty response

**Review by ShaileshParmar11 — COMMENTED** [review]
left few comments

**Comment by Sachin-chaurasiya** [comment]
> left few comments

Thanks, @ShaileshParmar11, I will push the changes.

**Review by ShaileshParmar11 — APPROVED** [review]
LGTM!

**Pull request merged**